src/devices/Seesaw/SeesawGpioDriver.cs

```diff
@@ -165,9 +169,13 @@ protected override PinValue Read(int pinNumber)
                 throw new InvalidOperationException("Can not read a value from a pin that is not open.");
             }
 
-            return _seesawDevice.ReadGpioDigital((byte)pinNumber) ? PinValue.High : PinValue.Low;
+            _pinValues[pinNumber] = _seesawDevice.ReadGpioDigital((byte)pinNumber) ? PinValue.High : PinValue.Low;
+            return _pinValues[pinNumber];
         }
 
+        /// <inheritdoc/>
+        protected override void Toggle(int pinNumber) => Write(pinNumber, !_pinValues[pinNumber]);
```

Comment:
I'm not sure dictionary actually makes it faster for those low level drivers - reading/writing pin is supposedely fastest operation you can make if you think about it. It might be currently faster this way because of OS but in principal that shouldn't be the case